smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelUpgrader.java

```diff
@@ -150,6 +158,12 @@ private boolean shouldV1MemberHaveDefaultTrait(ShapeType containerType, MemberSh
             && !target.hasTrait(BoxTrait.class);
     }
 
+    private boolean isDefaultPayload(Model model, MemberShape member, Shape target) {
+        return member.hasTrait(HttpPayloadTrait.class)
+                && member.getMemberTrait(model, StreamingTrait.class).isPresent()
```

Comment:
`@streaming` can only be on the `target` and can only target `blob` for defaulting purposes, simplifying this a little.